File: Whitelist/ContractDefinition/WhitelistDefinition.vb
Imports System
Imports System.Threading.Tasks
Imports System.Collections.Generic
Imports System.Numerics
Imports Nethereum.Hex.HexTypes
Imports Nethereum.ABI.FunctionEncoding.Attributes
Imports Nethereum.Web3
Imports Nethereum.RPC.Eth.DTOs
Imports Nethereum.Contracts.CQS
Imports Nethereum.Contracts
Imports System.Threading
Namespace White.Contracts.Whitelist.ContractDefinition

    
    
    Public Partial Class WhitelistDeployment
     Inherits WhitelistDeploymentBase
    
        Public Sub New()
            MyBase.New(DEFAULT_BYTECODE)
        End Sub
        
        Public Sub New(ByVal byteCode As String)
            MyBase.New(byteCode)
        End Sub
    
    End Class

    Public Class WhitelistDeploymentBase 
            Inherits ContractDeploymentMessage
        
        Public Shared DEFAULT_BYTECODE As String = "608060405234801561001057600080fd5b506040516104c33803806104c383398101604081905261002f91610049565b600280546001600160a01b03191633179055600055610062565b60006020828403121561005b57600080fd5b5051919050565b610452806100716000396000f3fe608060405234801561001057600080fd5b50600436106100935760003560e01c8063893d20e811610066578063893d20e81461010d5780638cd5a0b91461012857806391b173491461013b578063921a5b4814610098578063ba4365471461014357600080fd5b80633af32abf1461009857806358ef6747146100d957806372353033146100ee5780637a662f8f14610105575b600080fd5b6100c46100a63660046103aa565b6001600160a01b031660009081526004602052604090205460ff1690565b60405190151581526020015b60405180910390f35b6100ec6100e73660046103aa565b61014c565b005b6100f760015481565b6040519081526020016100d0565b6001546100f7565b6002546040516001600160a01b0390911681526020016100d0565b6100ec6101363660046103aa565b6102b5565b6000546100f7565b6100f760005481565b6002546001600160a01b031633146101ab5760405162461bcd60e51b815260206004820152601e60248201527f4572726f723a2043616c6c6572206973206e6f7420746865206f776e6572000060448201526064015b60405180910390fd5b6001600160a01b03811660009081526004602052604090205460ff16156102235760405162461bcd60e51b815260206004820152602660248201527f4572726f723a2053656e64657220616c7265616479206265656e2077686974656044820152651b1a5cdd195960d21b60648201526084016101a2565b600054600154106102765760405162461bcd60e51b815260206004820152601f60248201527f4572726f723a2057686974656c697374204c696d69742065786365656465640060448201526064016101a2565b6001600160a01b0381166000908152600460205260408120805460ff191660019081179091558054909182916102ad9083906103f0565b909155505050565b6002546001600160a01b0316331461030f5760405162461bcd60e51b815260206004820152601e60248201527f4572726f723a2043616c6c6572206973206e6f7420746865206f776e6572000060448201526064016101a2565b6001600160a01b03811660009081526004602052604090205460ff166103775760405162461bcd60e51b815260206004820181905260248201527f4572726f723a2053656e646572206973206e6f742077686974656c697374656460448201526064016101a2565b6001600160a01b0381166000908152600460205260408120805460ff1916905560018054909182916102ad908390610409565b6000602082840312156103bc57600080fd5b81356001600160a01b03811681146103d357600080fd5b9392505050565b634e487b7160e01b600052601160045260246000fd5b80820180821115610403576104036103da565b92915050565b81810381811115610403576104036103da56fea2646970667358221220ee29164f2c5289c149d90b7634040b2e542879e809479a050125ef7c2942bfbb64736f6c63430008130033"
        
        Public Sub New()
            MyBase.New(DEFAULT_BYTECODE)
        End Sub
        
        Public Sub New(ByVal byteCode As String)
            MyBase.New(byteCode)
        End Sub
        
        <[Parameter]("uint256", "_maxWhitelistedAddresses", 1)>
        Public Overridable Property [MaxWhitelistedAddresses] As BigInteger
    
    End Class    
    
    Public Partial Class AddUserAddressToWhitelistFunction
        Inherits AddUserAddressToWhitelistFunctionBase
    End Class

        <[Function]("addUserAddressToWhitelist")>
    Public Class AddUserAddressToWhitelistFunctionBase
        Inherits FunctionMessage
    
        <[Parameter]("address", "_addressToWhitelist", 1)>
        Public Overridable Property [AddressToWhitelist] As String
    
    End Class
    
    
    Public Partial Class GetMaxNumberOfWhitelistedAddressesFunction
        Inherits GetMaxNumberOfWhitelistedAddressesFunctionBase
    End Class

        <[Function]("getMaxNumberOfWhitelistedAddresses", "uint256")>
    Public Class GetMaxNumberOfWhitelistedAddressesFunctionBase
        Inherits FunctionMessage
    

    
    End Class
    
    
    Public Partial Class GetNumberOfWhitelistedAddressesFunction
        Inherits GetNumberOfWhitelistedAddressesFunctionBase
    End Class

        <[Function]("getNumberOfWhitelistedAddresses", "uint256")>
    Public Class GetNumberOfWhitelistedAddressesFunctionBase
        Inherits FunctionMessage
    

    
    End Class
    
    
    Public Partial Class GetOwnerFunction
        Inherits GetOwnerFunctionBase
    End Class

        <[Function]("getOwner", "address")>
    Public Class GetOwnerFunctionBase
        Inherits FunctionMessage
    

    
    End Class
    
    
    Public Partial Class IsWhitelistedFunction
        Inherits IsWhitelistedFunctionBase
    End Class

        <[Function]("isWhitelisted", "bool")>
    Public Class IsWhitelistedFunctionBase
        Inherits FunctionMessage
    
        <[Parameter]("address", "_whitelistedAddress", 1)>
        Public Overridable Property [WhitelistedAddress] As String
    
    End Class
    
    
    Public Partial Class MaxNumberOfWhitelistedAddressesFunction
        Inherits MaxNumberOfWhitelistedAddressesFunctionBase
    End Class

        <[Function]("maxNumberOfWhitelistedAddresses", "uint256")>
    Public Class MaxNumberOfWhitelistedAddressesFunctionBase
        Inherits FunctionMessage
    

    
    End Class
    
    
    Public Partial Class NumberOfAddressesWhitelistedFunction
        Inherits NumberOfAddressesWhitelistedFunctionBase
    End Class

        <[Function]("numberOfAddressesWhitelisted", "uint256")>
    Public Class NumberOfAddressesWhitelistedFunctionBase
        Inherits FunctionMessage
    

    
    End Class
    
    
    Public Partial Class RemoveUserAddressFromWhitelistFunction
        Inherits RemoveUserAddressFromWhitelistFunctionBase
    End Class

        <[Function]("removeUserAddressFromWhitelist")>
    Public Class RemoveUserAddressFromWhitelistFunctionBase
        Inherits FunctionMessage
    
        <[Parameter]("address", "_addressToRemove", 1)>
        Public Overridable Property [AddressToRemove] As String
    
    End Class
    
    
    Public Partial Class VerifyUserAddressFunction
        Inherits VerifyUserAddressFunctionBase
    End Class

        <[Function]("verifyUserAddress", "bool")>
    Public Class VerifyUserAddressFunctionBase
        Inherits FunctionMessage
    
        <[Parameter]("address", "_whitelistedAddress", 1)>
        Public Overridable Property [WhitelistedAddress] As String
    
    End Class
    
    
    
    
    Public Partial Class GetMaxNumberOfWhitelistedAddressesOutputDTO
        Inherits GetMaxNumberOfWhitelistedAddressesOutputDTOBase
    End Class

    <[FunctionOutput]>
    Public Class GetMaxNumberOfWhitelistedAddressesOutputDTOBase
        Implements IFunctionOutputDTO
        
        <[Parameter]("uint256", "", 1)>
        Public Overridable Property [ReturnValue1] As BigInteger
    
    End Class    
    
    Public Partial Class GetNumberOfWhitelistedAddressesOutputDTO
        Inherits GetNumberOfWhitelistedAddressesOutputDTOBase
    End Class

    <[FunctionOutput]>
    Public Class GetNumberOfWhitelistedAddressesOutputDTOBase
        Implements IFunctionOutputDTO
        
        <[Parameter]("uint256", "", 1)>
        Public Overridable Property [ReturnValue1] As BigInteger
    
    End Class    
    
    Public Partial Class GetOwnerOutputDTO
        Inherits GetOwnerOutputDTOBase
    End Class

    <[FunctionOutput]>
    Public Class GetOwnerOutputDTOBase
        Implements IFunctionOutputDTO
        
        <[Parameter]("address", "", 1)>
        Public Overridable Property [ReturnValue1] As String
    
    End Class    
    
    Public Partial Class IsWhitelistedOutputDTO
        Inherits IsWhitelistedOutputDTOBase
    End Class

    <[FunctionOutput]>
    Public Class IsWhitelistedOutputDTOBase
        Implements IFunctionOutputDTO
        
        <[Parameter]("bool", "", 1)>
        Public Overridable Property [ReturnValue1] As Boolean
    
    End Class    
    
    Public Partial Class MaxNumberOfWhitelistedAddressesOutputDTO
        Inherits MaxNumberOfWhitelistedAddressesOutputDTOBase
    End Class

    <[FunctionOutput]>
    Public Class MaxNumberOfWhitelistedAddressesOutputDTOBase
        Implements IFunctionOutputDTO
        
        <[Parameter]("uint256", "", 1)>
        Public Overridable Property [ReturnValue1] As BigInteger
    
    End Class    
    
    Public Partial Class NumberOfAddressesWhitelistedOutputDTO
        Inherits NumberOfAddressesWhitelistedOutputDTOBase
    End Class

    <[FunctionOutput]>
    Public Class NumberOfAddressesWhitelistedOutputDTOBase
        Implements IFunctionOutputDTO
        
        <[Parameter]("uint256", "", 1)>
        Public Overridable Property [ReturnValue1] As BigInteger
    
    End Class    
    
    
    
    Public Partial Class VerifyUserAddressOutputDTO
        Inherits VerifyUserAddressOutputDTOBase
    End Class

    <[FunctionOutput]>
    Public Class VerifyUserAddressOutputDTOBase
        Implements IFunctionOutputDTO
        
        <[Parameter]("bool", "", 1)>
        Public Overridable Property [ReturnValue1] As Boolean
    
    End Class
End Namespace

File: Whitelist/ContractDefinition/WhitelistDefinition.fs
namespace White.Contracts.Whitelist.ContractDefinition

open System
open System.Threading.Tasks
open System.Collections.Generic
open System.Numerics
open Nethereum.Hex.HexTypes
open Nethereum.ABI.FunctionEncoding.Attributes
open Nethereum.Web3
open Nethereum.RPC.Eth.DTOs
open Nethereum.Contracts.CQS
open Nethereum.Contracts
open System.Threading

    
    
    type WhitelistDeployment(byteCode: string) =
        inherit ContractDeploymentMessage(byteCode)
        
        static let BYTECODE = "608060405234801561001057600080fd5b506040516106e83803806106e883398101604081905261002f91610049565b600280546001600160a01b03191633179055600055610062565b60006020828403121561005b57600080fd5b5051919050565b610677806100716000396000f3fe608060405234801561001057600080fd5b506004361061009e5760003560e01c806391b173491161006657806391b173491461010a578063921a5b4814610112578063b0c8f9dc1461014e578063ba43654714610161578063bb9c6c3e1461016a57600080fd5b806358ef6747146100a357806372353033146100b85780637a662f8f146100d4578063893d20e8146100dc5780638cd5a0b9146100f7575b600080fd5b6100b66100b1366004610494565b61017d565b005b6100c160015481565b6040519081526020015b60405180910390f35b6001546100c1565b6002546040516001600160a01b0390911681526020016100cb565b6100b6610105366004610494565b6102e6565b6000546100c1565b61013e610120366004610494565b6001600160a01b031660009081526004602052604090205460ff1690565b60405190151581526020016100cb565b6100c161015c3660046104da565b6103db565b6100c160005481565b6100c16101783660046104da565b61043b565b6002546001600160a01b031633146101dc5760405162461bcd60e51b815260206004820152601e60248201527f4572726f723a2043616c6c6572206973206e6f7420746865206f776e6572000060448201526064015b60405180910390fd5b6001600160a01b03811660009081526004602052604090205460ff16156102545760405162461bcd60e51b815260206004820152602660248201527f4572726f723a2053656e64657220616c7265616479206265656e2077686974656044820152651b1a5cdd195960d21b60648201526084016101d3565b600054600154106102a75760405162461bcd60e51b815260206004820152601f60248201527f4572726f723a2057686974656c697374204c696d69742065786365656465640060448201526064016101d3565b6001600160a01b0381166000908152600460205260408120805460ff191660019081179091558054909182916102de9083906105a1565b909155505050565b6002546001600160a01b031633146103405760405162461bcd60e51b815260206004820152601e60248201527f4572726f723a2043616c6c6572206973206e6f7420746865206f776e6572000060448201526064016101d3565b6001600160a01b03811660009081526004602052604090205460ff166103a85760405162461bcd60e51b815260206004820181905260248201527f4572726f723a2053656e646572206973206e6f742077686974656c697374656460448201526064016101d3565b6001600160a01b0381166000908152600460205260408120805460ff1916905560018054909182916102de9083906105ba565b3360008181526004602052604081205490919060ff1661040d5760405162461bcd60e51b81526004016101d3906105cd565b6000429050806003856040516104239190610612565b90815260405190819003602001902055915050919050565b3360008181526004602052604081205490919060ff1661046d5760405162461bcd60e51b81526004016101d3906105cd565b60038360405161047d9190610612565b908152602001604051809103902054915050919050565b6000602082840312156104a657600080fd5b81356001600160a01b03811681146104bd57600080fd5b9392505050565b634e487b7160e01b600052604160045260246000fd5b6000602082840312156104ec57600080fd5b813567ffffffffffffffff8082111561050457600080fd5b818401915084601f83011261051857600080fd5b81358181111561052a5761052a6104c4565b604051601f8201601f19908116603f01168101908382118183101715610552576105526104c4565b8160405282815287602084870101111561056b57600080fd5b826020860160208301376000928101602001929092525095945050505050565b634e487b7160e01b600052601160045260246000fd5b808201808211156105b4576105b461058b565b92915050565b818103818111156105b4576105b461058b565b60208082526025908201527f57686974656c6973743a20596f75206e65656420746f2062652077686974656c6040820152641a5cdd195960da1b606082015260800190565b6000825160005b818110156106335760208186018101518583015201610619565b50600092019182525091905056fea26469706673582212201160474853e3b8c8ee0fe819ece1615298dd937b13cd7682482d1b7e9b0061b864736f6c63430008130033"
        
        new() = WhitelistDeployment(BYTECODE)
        
            [<Parameter("uint256", "_maxWhitelistedAddresses", 1)>]
            member val public MaxWhitelistedAddresses = Unchecked.defaultof<BigInteger> with get, set
        
    
    [<Function("add", "uint256")>]
    type AddFunction() = 
        inherit FunctionMessage()
    
            [<Parameter("string", "hash", 1)>]
            member val public Hash = Unchecked.defaultof<string> with get, set
        
    
    [<Function("addUserAddressToWhitelist")>]
    type AddUserAddressToWhitelistFunction() = 
        inherit FunctionMessage()
    
            [<Parameter("address", "_addressToWhitelist", 1)>]
            member val public AddressToWhitelist = Unchecked.defaultof<string> with get, set
        
    
    [<Function("getMaxNumberOfWhitelistedAddresses", "uint256")>]
    type GetMaxNumberOfWhitelistedAddressesFunction() = 
        inherit FunctionMessage()
    

        
    
    [<Function("getNumberOfWhitelistedAddresses", "uint256")>]
    type GetNumberOfWhitelistedAddressesFunction() = 
        inherit FunctionMessage()
    

        
    
    [<Function("getOwner", "address")>]
    type GetOwnerFunction() = 
        inherit FunctionMessage()
    

        
    
    [<Function("maxNumberOfWhitelistedAddresses", "uint256")>]
    type MaxNumberOfWhitelistedAddressesFunction() = 
        inherit FunctionMessage()
    

        
    
    [<Function("numberOfAddressesWhitelisted", "uint256")>]
    type NumberOfAddressesWhitelistedFunction() = 
        inherit FunctionMessage()
    

        
    
    [<Function("removeUserAddressFromWhitelist")>]
    type RemoveUserAddressFromWhitelistFunction() = 
        inherit FunctionMessage()
    
            [<Parameter("address", "_addressToRemove", 1)>]
            member val public AddressToRemove = Unchecked.defaultof<string> with get, set
        
    
    [<Function("verify", "uint256")>]
    type VerifyFunction() = 
        inherit FunctionMessage()
    
            [<Parameter("string", "hash", 1)>]
            member val public Hash = Unchecked.defaultof<string> with get, set
        
    
    [<Function("verifyUserAddress", "bool")>]
    type VerifyUserAddressFunction() = 
        inherit FunctionMessage()
    
            [<Parameter("address", "_whitelistedAddress", 1)>]
            member val public WhitelistedAddress = Unchecked.defaultof<string> with get, set
        
    
    
    
    
    
    [<FunctionOutput>]
    type GetMaxNumberOfWhitelistedAddressesOutputDTO() =
        inherit FunctionOutputDTO() 
            [<Parameter("uint256", "", 1)>]
            member val public ReturnValue1 = Unchecked.defaultof<BigInteger> with get, set
        
    
    [<FunctionOutput>]
    type GetNumberOfWhitelistedAddressesOutputDTO() =
        inherit FunctionOutputDTO() 
            [<Parameter("uint256", "", 1)>]
            member val public ReturnValue1 = Unchecked.defaultof<BigInteger> with get, set
        
    
    [<FunctionOutput>]
    type GetOwnerOutputDTO() =
        inherit FunctionOutputDTO() 
            [<Parameter("address", "", 1)>]
            member val public ReturnValue1 = Unchecked.defaultof<string> with get, set
        
    
    [<FunctionOutput>]
    type MaxNumberOfWhitelistedAddressesOutputDTO() =
        inherit FunctionOutputDTO() 
            [<Parameter("uint256", "", 1)>]
            member val public ReturnValue1 = Unchecked.defaultof<BigInteger> with get, set
        
    
    [<FunctionOutput>]
    type NumberOfAddressesWhitelistedOutputDTO() =
        inherit FunctionOutputDTO() 
            [<Parameter("uint256", "", 1)>]
            member val public ReturnValue1 = Unchecked.defaultof<BigInteger> with get, set
        
    
    
    
    [<FunctionOutput>]
    type VerifyOutputDTO() =
        inherit FunctionOutputDTO() 
            [<Parameter("uint256", "dateAdded", 1)>]
            member val public DateAdded = Unchecked.defaultof<BigInteger> with get, set
        
    
    [<FunctionOutput>]
    type VerifyUserAddressOutputDTO() =
        inherit FunctionOutputDTO() 
            [<Parameter("bool", "", 1)>]
            member val public ReturnValue1 = Unchecked.defaultof<bool> with get, set
    

